src/ansi-c/expr2cleanc.h

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************\
+
+Module:
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
+/// \file
+/// expr2cleanct
+
+#ifndef CPROVER_ANSI_C_EXPR2CLEANC_H
+#define CPROVER_ANSI_C_EXPR2CLEANC_H
+
+#include "expr2c_class.h"
+
+#include <string>
+
+/// Produces C from expression and types.
+/// It does not print padding components in structs
+/// It does not print array sizes when printing array types
+/// It does not print the body of the struct when printing the struct type
+class expr2cleanct:public expr2ct
```

Comment:
I don't see much value in having this as a separate class. I'd prefer to see the behaviour of expr2ct be made configurable.

Reply:
Did you see this branch: https://github.com/thk123/cbmc/tree/dump/expr2c-configuration. It gets complicated as TG uses a broader interface than just `expr2c/type2c` so the options are:
1. rework TG to use the regular interface (probably at least a couple of days)
2. add an extra method in `expr2c.h` for this extra route (half day?)
3. just add the extra method to `expr2ct` and use the class directly (quickest - but I think have the class defined in a separate header is supposed to indicate it _isn't_ part of the public interface so feels like the worst option)

I'll have a look at the second one unless you think the third one is acceptable?

Reply:
I have had a brief look at the configuration branch and really liked it! Given my limited insight: can you briefly explain what this "broader interface" is that TG uses/requires? Looking at dump-c, this work will help simplify its code a lot, which of course I'll take on once any approach here is merged. I do suspect that TG may have similar pieces of code?

Reply:
I missed that one. Yes, @thk123, that's what I had envisaged.